src/Illuminate/Validation/Validator.php

```diff
@@ -2262,6 +2279,64 @@ protected function normalizeRule($rule)
     }
 
     /**
+     * Determine if the given rule accepts other field names as parameters.
+     * 
+     * @param  string  $rule
+     * @return bool
+     */
+    protected function acceptsFields($rule)
+    {
+        return in_array($rule, $this->fieldRules);
+    }
+
+    /**
+     * Get the numeric keys from an attribute flattened with dot notation.
+     *
+     * E.g. 'foo.1.bar.2.baz' -> [1, 2]
+     *
+     * @param  string  $attribute
+     * @return array
+     */
+    protected function getNumericKeys($attribute)
+    {
+        if (preg_match_all('/\.(\d+)\./', $attribute, $keys)) {
+            array_shift($keys);
+
+            return Arr::collapse($keys);
+        }
+
+        return [];
+    }
```

Comment:
`$keys` is an array of arrays, you removed the unwanted `$keys[0]` then collapsed the remaining of `$keys` (so, `$keys[1]`) to get a flat array.

Here, I directly pick `$keys[1]` which is already a flat array. You don't have to (and cannot) collapse it.